pkg/controller/operators/olm/apiservices.go

```diff
@@ -0,0 +1,428 @@
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: secret.GetName(),
+				Items: []corev1.KeyToPath{
+					{
+						Key:  "tls.crt",
+						Path: "apiserver.crt",
+					},
+					{
+						Key:  "tls.key",
+						Path: "apiserver.key",
+					},
+				},
+			},
+		},
+	}
+
+	replaced := false
+	for i, v := range depSpec.Template.Spec.Volumes {
+		if v.Name == volume.Name {
+			depSpec.Template.Spec.Volumes[i] = volume
+			replaced = true
+			break
+		}
+	}
+	if !replaced {
+		depSpec.Template.Spec.Volumes = append(depSpec.Template.Spec.Volumes, volume)
+	}
+
+	// TODO(NICK): limit what containers get a volume mount for the certs volume
+	// mount to the default serving cert path
+	mount := corev1.VolumeMount{
+		Name:      volume.Name,
+		MountPath: "/apiserver.local.config/certificates",
+	}
+	for i, container := range depSpec.Template.Spec.Containers {
+		found := false
+		for j, m := range container.VolumeMounts {
+			if m.Name == mount.Name {
+				found = true
+				break
+			}
+
+			// replace if mounting to the same location
+			if m.MountPath == mount.MountPath {
+				container.VolumeMounts[j] = mount
+				found = true
+				break
+			}
+		}
+		if !found {
+			container.VolumeMounts = append(container.VolumeMounts, mount)
+		}
+
+		depSpec.Template.Spec.Containers[i] = container
+	}
+
+	// create APIService with fresh CA bundle
+	caCertPEM, _, err := ca.ToPEM()
+	if err != nil {
+		logger.Debugf("unable to convert CA certificate to PEM format for APIService %s", apiServiceName)
+		return nil, err
+	}
+
+	apiService := &apiregistrationv1.APIService{
+		Spec: apiregistrationv1.APIServiceSpec{
+			Service: &apiregistrationv1.ServiceReference{
+				Namespace: service.GetNamespace(),
+				Name:      service.GetName(),
+			},
+			Group:    desc.Group,
+			Version:  desc.Version,
+			CABundle: caCertPEM,
+			// TODO(Nick): are hardcoded priorities okay?
+			GroupPriorityMinimum: int32(2000),
+			VersionPriority:      int32(15),
+		},
+	}
+	apiService.SetName(apiServiceName)
+	ownerutil.AddNonBlockingOwner(apiService, csv)
+
+	_, err = a.OpClient.CreateAPIService(apiService)
+	if k8serrors.IsAlreadyExists(err) {
+		// attempt a replace
+		deleteErr := a.OpClient.DeleteAPIService(apiService.GetName(), &metav1.DeleteOptions{})
+		if _, err := a.OpClient.CreateAPIService(apiService); err != nil || deleteErr != nil {
+			logger.Debugf("could not replace APIService %s", apiService.GetName())
+			return nil, err
+		}
+	} else if err != nil {
+		log.Debugf("could not create APIService %s", apiService.GetName())
+		return nil, err
+	}
+
+	// change depSpec.Template's name to force rollout
+	// ensures that fresh secret is used by apiserver if not hot reloading
+	podTemplateName := names.SimpleNameGenerator.GenerateName(apiServiceName)
+	if podTemplateName == depSpec.Template.GetName() {
+		return nil, fmt.Errorf("a name collision occured when generating name for PodTemplate")
+	}
+	depSpec.Template.SetName(podTemplateName)
```

Comment:
This ensures that the extension api-server is actually rolled out in the event of a secret change. Triggering rollouts this way allows us to keep a static secret name.